File: java/test/Ice/objects/UnexpectedObjectExceptionTestI.java
// **********************************************************************
//
// Copyright (c) 2003-2007 ZeroC, Inc. All rights reserved.
//
// This copy of Ice is licensed to you under the terms described in the
// ICE_LICENSE file included in this distribution.
//
// **********************************************************************

import Test.*;

public final class UnexpectedObjectExceptionTestI extends Ice.Blobject
{
    public boolean
    ice_invoke(byte[] inParams, Ice.ByteSeqHolder outParams, Ice.Current current)
    {
        Ice.Communicator communicator = current.adapter.getCommunicator();
        Ice.OutputStream out = Ice.Util.createOutputStream(communicator);
        AlsoEmpty ae = new AlsoEmpty();
        Test.AlsoEmptyHelper.write(out, ae);
        out.writePendingObjects();
        outParams.value = out.finished();
        return true;
    }
}

File: csharp/test/Ice/objects/UnexpectedObjectExceptionTestI.cs
// Copyright (c) ZeroC, Inc.

namespace Ice.objects
{
        public sealed class UnexpectedObjectExceptionTestI : Ice.Blobject
        {
            public override bool ice_invoke(byte[] inParams, out byte[] outParams, Ice.Current current)
            {
                var communicator = current.adapter.getCommunicator();
                var @out = new Ice.OutputStream(communicator);
                @out.startEncapsulation(current.encoding, FormatType.SlicedFormat);
                var ae = new Test.AlsoEmpty();
                @out.writeValue(ae);
                @out.writePendingValues();
                @out.endEncapsulation();
                outParams = @out.finished();
                return true;
            }
        }
    }

